streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -240,8 +240,8 @@
                     STATE_DIR_DOC)
             .define(REPLICATION_FACTOR_CONFIG,
                     Type.INT,
-                    1,
-                    Importance.MEDIUM,
+                    3,
```

Comment:
I think this dynamic change is quite dangerous -- if I specify replication 3 and cannot get it, I want an exception... Thus, I would leave replication factor to 1 for demoing purpose -- if anyone goes to production she can set to whatever value is suitable -- or we make the parameter non-optional.

I think it would be a hassle to have default value 3, and overwrite it with 1 in each example we do...